File: 프로그래머스/2/70129. 이진 변환 반복하기/이진 변환 반복하기.cpp
#include <string>
#include <vector>
#include <bits/stdc++.h>

using namespace std;

vector<int> solution(string s) {
    vector<int> answer;
    int zero = 0;
    int time = 0;
    
    while(s != "1") {
        string no_zero_s = "";
        for(int i=0; i<s.size(); i++){
            if(s[i] == '0') zero++;
            else no_zero_s += '1';
        }
        
        int c = no_zero_s.size();
        
        string new_s = "";
        
        while(c != 0) {
            new_s += to_string(c % 2);
            c /= 2;
        }
        
        reverse(new_s.begin(), new_s.end());
        s = new_s;
        time++;
    }
    
    answer.push_back(time);
    answer.push_back(zero);
    
    return answer;
}

File: 프로그래머스/2/70129. 이진 변환 반복하기/이진 변환 반복하기.java
import java.util.*;

class Solution {
    public int[] solution(String s) {
        int[] answer = {};
        
        int zero = 0;
        int time = 0;
        
        StringBuilder sb = new StringBuilder();
        
        while(!s.equals("1")) {
            sb.setLength(0);
            
            for(int i=0; i<s.length(); i++){
                if (s.charAt(i) == '0') {
                    zero++;
                }
                else {
                    sb.append('1');
                }
            }
            
            int c = sb.length();
            
            s = Integer.toBinaryString(c);
            time++;
        }

        return new int[]{time, zero};
    }
}